lib/aruco_detector.hpp

```diff
@@ -1,55 +1,69 @@
+#pragma once
+
 #include <opencv2/opencv.hpp>
 #include <opencv2/aruco.hpp>
-#include <memory>
-#include "classification_model.hpp"
+#include <iostream>
+#include <vector>
+
+struct ArUcoDetectionResult {
+    std::vector<int> ids;
+    std::vector<std::vector<cv::Point2f>> corners;
+    std::vector<cv::Vec3d> rvecs;
+    std::vector<cv::Vec3d> tvecs;
+};
 
-class Aruco_Detector : public ClassificationModel::dClassification {
+class Aruco_Detector {
 public:
-    Aruco_Detector(cv::VideoCapture& videoCapture, int dictionaryType) : inputVideo(videoCapture), dictionaryType(dictionaryType) {}
+    Aruco_Detector(const cv::Mat& cameraMatrix, int dictionaryType,)
+            : cameraMatrix(cameraMatrix), dictionaryType(dictionaryType) {}
 
-    ClassificationModel::Detection classify(cv::Mat& image, float threshold) override {
+    enum Detection {
+        NONE = 0,
+        ARUCO
+    };
+
+    Detection classify(cv::Mat& image, float threshold) {
```

Comment:
this method should be `friend` and the name should be `model_classify`. Refer: https://github.com/kknives/michi/blob/fd030796cc2bcc88a81b5c568af13e30245abc91/lib/yolov8_arrow.hpp#L166